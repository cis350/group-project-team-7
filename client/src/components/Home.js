import React from "react";

const Home = () => {
	return (
		<>
			<div className="relative overflow-hidden bg-white">
				<div className="mx-auto max-w-7xl">
					<div className="relative z-10 bg-white pb-8 sm:pb-16 md:pb-20 lg:w-full lg:max-w-2xl lg:pb-28 xl:pb-32">
						<svg
							className="absolute inset-y-0 right-0 hidden h-full w-48 translate-x-1/2 transform text-white lg:block"
							fill="currentColor"
							viewBox="0 0 100 100"
							preserveAspectRatio="none"
							aria-hidden="true"
						>
							<polygon points="50,0 100,0 50,100 0,100" />
						</svg>

						<main className="mx-auto mt-10 max-w-7xl px-6 sm:mt-12 md:mt-16 lg:mt-20 lg:px-8 xl:mt-28">
							<div className="sm:text-center lg:text-left">
								<h1 className="text-4xl font-bold tracking-tight text-gray-900 sm:text-5xl md:text-5xl">
									<span className="block">Moder Patshala:</span>{" "}
									<span className="block text-mainBlue">
										Guiding Students and Parents to a Brighter Future Since 2004
									</span>
								</h1>
								<div className="mt-5 sm:mt-8 sm:flex sm:justify-center lg:justify-start">
									<div className="rounded-md shadow">
										<a
											href="/login"
											className="flex w-full items-center justify-center rounded-md border border-transparent bg-mainBlue px-8 py-3 text-base font-medium text-white hover:bg-darkBlue md:py-4 md:px-10 md:text-lg"
										>
											Login
										</a>
									</div>
									<div className="mt-3 sm:mt-0 sm:ml-3">
										<a
											href="/signup"
											className="flex w-full items-center justify-center rounded-md border border-transparent bg-mainPeach px-8 py-3 text-base font-medium text-white hover:bg-darkPeach md:py-4 md:px-10 md:text-lg"
										>
											Signup
										</a>
									</div>
								</div>
							</div>
						</main>
					</div>
				</div>
			</div>
		</>
	);
};

export default Home;
