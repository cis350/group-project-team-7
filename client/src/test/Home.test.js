import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from '../components/Home.js';

describe('Home Component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders the main headline', () => {
    render(<Home />);
    const headline = screen.getByText(/Guiding Students and Parents to a Brighter Future Since 2004/i);
    expect(headline).toBeInTheDocument();
  });

  test('has a login link that navigates to the login page', () => {
    render(<Home />);
    const loginLink = screen.getByRole('link', { name: /login/i });
    expect(loginLink).toBeInTheDocument();
    expect(loginLink).toHaveAttribute('href', '/login');
  });

  test('has a signup link that navigates to the signup page', () => {
    render(<Home />);
    const signupLink = screen.getByRole('link', { name: /signup/i });
    expect(signupLink).toBeInTheDocument();
    expect(signupLink).toHaveAttribute('href', '/signup');
  });
});
