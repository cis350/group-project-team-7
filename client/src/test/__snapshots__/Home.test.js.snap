// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden bg-white"
  >
    <div
      class="mx-auto max-w-7xl"
    >
      <div
        class="relative z-10 bg-white pb-8 sm:pb-16 md:pb-20 lg:w-full lg:max-w-2xl lg:pb-28 xl:pb-32"
      >
        <svg
          aria-hidden="true"
          class="absolute inset-y-0 right-0 hidden h-full w-48 translate-x-1/2 transform text-white lg:block"
          fill="currentColor"
          preserveAspectRatio="none"
          viewBox="0 0 100 100"
        >
          <polygon
            points="50,0 100,0 50,100 0,100"
          />
        </svg>
        <main
          class="mx-auto mt-10 max-w-7xl px-6 sm:mt-12 md:mt-16 lg:mt-20 lg:px-8 xl:mt-28"
        >
          <div
            class="sm:text-center lg:text-left"
          >
            <h1
              class="text-4xl font-bold tracking-tight text-gray-900 sm:text-5xl md:text-5xl"
            >
              <span
                class="block"
              >
                Moder Patshala:
              </span>
               
              <span
                class="block text-mainBlue"
              >
                Guiding Students and Parents to a Brighter Future Since 2004
              </span>
            </h1>
            <div
              class="mt-5 sm:mt-8 sm:flex sm:justify-center lg:justify-start"
            >
              <div
                class="rounded-md shadow"
              >
                <a
                  class="flex w-full items-center justify-center rounded-md border border-transparent bg-mainBlue px-8 py-3 text-base font-medium text-white hover:bg-darkBlue md:py-4 md:px-10 md:text-lg"
                  href="/login"
                >
                  Login
                </a>
              </div>
              <div
                class="mt-3 sm:mt-0 sm:ml-3"
              >
                <a
                  class="flex w-full items-center justify-center rounded-md border border-transparent bg-mainPeach px-8 py-3 text-base font-medium text-white hover:bg-darkPeach md:py-4 md:px-10 md:text-lg"
                  href="/signup"
                >
                  Signup
                </a>
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
